### DISCUSSION TITLE
Posts templates

### DISCUSSION THREAD

**Discussion by jespino** [mattermost/community#140] [Enhancements]
Have a set of predefined templates with placeholders to reply to certain things. This is specially useful for the support team, I guess.

**Comment by chetanyakan** [comment]
We should also have the option to add user-defined templates.

**Reply by jespino** [reply]
Actually, I was thinking about that user-defined templates, but you are bringing an interesting thing, about shared templates in groups of people, maybe by channel membership or any other approach.

**Reply by chetanyakan** [reply]
So Interestingly, I just noticed GitHub has the option for reply templates (top right button: `insert a reply`)

![image](https://user-images.githubusercontent.com/35728906/123428173-292c6580-d5e3-11eb-83dd-22ac30d23427.png)

**Reply by chetanyakan** [reply]
> in groups of people, maybe by channel membership

This is an interesting idea. Using channel membership for custom roles.

Plugins could ask for a `channelID` and all members of that channel can have access to particular plugin features.

**Reply by esethna** [reply]
Yes user-defined or user group templates make a lot of sense, especially in channels like Announcements

**Reply by teresa-novoa-zz** [reply]
Great idea!

**Comment by Willyfrog** [comment]
this is something that we should have long ago, but it could definitely be conbined with the global drafts proposal: #63

**Reply by jespino** [reply]
I think draft and templates had to different purposes, and I don't think the combination work great. I think the global drafts is a great idea, but this tries to solve another problem, i think.
